source/spark_sql_migrations/spark_sql_migrations/infrastructure/apply_migration_scripts.py

```diff
@@ -0,0 +1,75 @@
+import pyspark.sql.functions as F
+import spark_sql_migrations.utility.delta_table_helper as delta_table_helper
+import spark_sql_migrations.infrastructure.sql_file_executor as sql_file_executor
+from typing import List
+from pyspark.sql import SparkSession
+from dependency_injector.wiring import Provide, inject
+from spark_sql_migrations.models.table_version import TableVersion
+from spark_sql_migrations.models.configuration import Configuration
+from spark_sql_migrations.container import SparkSqlMigrationsContainer
+
+
+def apply_migration_scripts(
+        uncommitted_migrations: List[str]
+) -> None:
+    _apply_migration_scripts(uncommitted_migrations)
+
+
+@inject
+def _apply_migration_scripts(
+        uncommitted_migrations: list[str],
+        config: Configuration = Provide[SparkSqlMigrationsContainer.config]
+) -> None:
+    for migration in uncommitted_migrations:
+        table_versions = _get_table_versions()
+        try:
```

Comment:
Does this mean that if we apply X migrations and it fails for instance at the last migration, then we will roll back only the last one? If yes, couldn't that be a problem for later PRs?

Reply:
I was wondering if we should roll back all migrations if one fails? What do you think?